do_mpc/controller.py

```diff
@@ -30,6 +30,666 @@
+        assert self.model.flags['dae2odemodel']==False, 'Already performing input rate penalization without the use of this function'
+        
+        #Verifying input cost matrix for input rate penalization
+        assert self.R_delu.size != 0 , 'set R_delu parameter using set_param() fun.'
+        
+        #Modifying A and B matrix for input rate penalization
+        identity_u = np.identity(np.shape(self.B)[1])
+        zeros_A = np.zeros((np.shape(self.B)[1],np.shape(self.A)[1]))
+        self.A_new = np.block([[self.A,self.B],[zeros_A,identity_u]])
+        self.B_new = np.block([[self.B],[identity_u]])
+        zeros_Q = np.zeros((np.shape(self.Q)[0],np.shape(self.R)[1]))
+        zeros_Ru = np.zeros((np.shape(self.R)[0],np.shape(self.Q)[1]))
+        
+        #Modifying Q and R matrix for input rate penalization
+        self.Q = np.block([[self.Q,zeros_Q],[zeros_Ru,self.R]])
+        self.P = np.block([[self.P,zeros_Q],[zeros_Ru,self.R]])
+        self.R = self.R_delu
+        
+        #Computing gain matrix
+        K = self.discrete_gain(self.A_new, self.B_new)
+        return K
+ 
+    
+    def set_param(self,**kwargs):
+        """Set the parameters of the :py:class:`LQR` class. Parameters must be passed as pairs of valid keywords and respective argument.
+        For example:
+
+        ::
+
+            lqr.set_param(n_horizon = 20)
+
+        It is also possible and convenient to pass a dictionary with multiple parameters simultaneously as shown in the following example:
+
+        ::
+
+            setup_lqr = {
+                'n_horizon': 20,
+                't_sample': 0.5,
+            }
+            lqr.set_param(**setup_mpc)
+        
+        This makes use of thy python "unpack" operator. See `more details here`_.
+
+        .. _`more details here`: https://codeyarns.github.io/tech/2012-04-25-unpack-operator-in-python.html
+
+        .. note:: The only required parameters  are ``n_horizon``. All other parameters are optional.
+
+        .. note:: :py:func:`set_param` can be called multiple times. Previously passed arguments are overwritten by successive calls.
+
+        The following parameters are available:
+            
+        :param n_horizon: Prediction horizon of the optimal control problem. Parameter must be set by user.
+        :type n_horizon: int
+
+        :param t_sample: Sampling time for converting continuous time system to discrete time system.
+        :type t_sample: float
+        
+        :param mode: mode for operating LQR
+        :type mode: String
+        
+        :param conv_method: Method for converting continuous time to discrete time system
+        :type conv_method: String
+        
+        """
+        for key, value in kwargs.items():
+            if not (key in self.data_fields):
+                print('Warning: Key {} does not exist for LQR.'.format(key))
+            else:
+                setattr(self, key, value)
+                        
+    def steady_state(self):
```

Comment:
Can we actually make this a method of the model? Knowing the steady state is actually great for nonlinear systems as well. For nonlinear systems we would need to formulate and optimization problem, however.